### ISSUE TITLE
Extend an already configured RetryPolicy 

### ISSUE THREAD

**Issue by austince** [failsafe-lib/failsafe#251]
Hi there,

I'd like to extend a `RetryPolicy` that's already been configured w/out throwing errors like `Delays have already been set`. 

```java
class BaseClient {
  @Override
  protected RetryPolicy<Object> getDefaultRetryPolicy() {
    return new RetryPolicy<>()
        .withBackoff(1, 30, ChronoUnit.SECONDS);
  }
}

class Client extends BaseClient {
  @Override
  protected RetryPolicy<Object> getDefaultRetryPolicy() {
    return super.getDefaultRetryPolicy()
        .withBackoff(1, 60, ChronoUnit.SECONDS); // throws "Delays have already been set"
  }
}
```

Is this already possible/ something the community would consider? Why is is currently an error to configure something multiple times?

**Comment by austince** [comment]
Related to https://github.com/jhalterman/failsafe/issues/254

**Comment by jhalterman** [comment]
The reason for the error currently isn't that configuring something multiple times is bad, it's intended to avoid confusion from configuring different types of delays. For example, if you configure a fixed delay, a backoff delay, and a random delay, how do you know which one is actually used? That's the purpose for this exception.

Perhaps we could relax things to allow configuring the same type of delay more than once, but not another type?

Separately, 3.0 is about to be released which introduces a RetryPolicy builder for configuring a RetryPolicy, but the same behavior is being carried over for now.

**Comment by whiskeysierra** [comment]
@jhalterman Now with 3.0 on the horizon, you could take a look whether another API (style) change could be applied which would tackle problems like this at compile time already: https://whiskeysierra.github.io/knowledge/patterns/staged-builders/

**Comment by jhalterman** [comment]
I just pushed a51a037 which relaxes some of the illegal state validation in `RetryPolicyBuilder` so that different types of delays can be configured to replace previous configuration, and relaxes the requirement that a jitter be configured after a delay.

**Comment by jhalterman** [comment]
I just released 3.0.1 with this change.
